fh-sync-js.d.ts

```diff
@@ -426,12 +426,20 @@ declare module SyncClient {
    * @param handler - function that wraps underlying storage solution
    */
   function setStorageAdapter(handler: (dataset_id: string, isSave: boolean, cb: any) => void): void;
+
   /**
    * Allows to override default hashing method
    *
    * @param method - function that wraps underlying hashing method
    */
   function setHashMethod(method: () => any): void;
+
+  /**
+   * Allows to override default method for checking if application is online
+   *
+   * @param method - function that checks if network is available
+   */
+  function setNetworkStatusHandler(method: (callback: (isOnline: boolean) => void) => void): void;
```

Comment:
This is typical implementation of `Strategy` pattern. We trying to pass here a function as argument of function (this is called delegate in most of the languages) and this function will be used as NetworkStatusHandler.